### PR TITLE
[FLINK-8475][config][docs] Integrate HA options

### PR DESCRIPTION
**Pull request by zentol**
## What is the purpose of the change

This PR integrates the HA `ConfigOptions` into the configuration docs generator.

## Brief change log

* Add missing descriptions to config options (derived from existing description/javadocs)
* integrate HA configuration table into `config.md`

### CODE REVIEW

File: docs/_includes/generated/high_availability_configuration.html [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            <th class="text-left" style="width: 20%">Key</th>
+            <th class="text-left" style="width: 15%">Default</th>
+            <th class="text-left" style="width: 65%">Description</th>
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td><h5>high-availability</h5></td>
+            <td>"NONE"</td>
+            <td>Defines high-availability mode used for the cluster execution. To enable high-availability, set this mode to "ZOOKEEPER".</td>
```
Comment (aljoscha) [inline docs/_includes/generated/high_availability_configuration.html]: Why do these not have all the description that the original doc has?

Comment (zentol) [reply docs/_includes/generated/high_availability_configuration.html]: high-availability: how Zookeeper-based HA works is/should be explained in the [HA docs](https://ci.apache.org/projects/flink/flink-docs-master/ops/jobmanager_high_availability.html#yarn-cluster-high-availability)
cluster-id: derived from javadocs, as the existing docs were entirely about ZK setups
job.delay: derived from javadocs, as i assume it to be more up-to-date than the ZK specific description
jobmanager.port: undocumented before
storageDir: derived from javadocs, as i assume it to be more up-to-date than the ZK specific description

**Comment by aljoscha** [comment]
Ok, then please go ahead.

**Comment by zentol** [comment]
merging

**Pull request closed without merging**